### PR TITLE
[AArch64] Cleanup existing values in getMemOpInfo

### PR DESCRIPTION
**Pull request by davemgreen**
This patch tries to clean up some of the existing values in getMemOpInfo. All values should now be in bytes (not bits), and the MinOffset/MaxOffset are now always represented unscaled (the immediate that will be present in the final instruction).

Although I could not find a place where it altered codegen, the offset of a post-index instruction will be 0, not scale*imm. A IsPostIndexLdStOpcode method has been added to try and make sure that case is handled properly.

### CODE REVIEW

File: llvm/lib/Target/AArch64/AArch64InstrInfo.cpp [hunk 1]
```diff
@@ -3731,17 +3954,17 @@ bool AArch64InstrInfo::getMemOpInfo(unsigned Opcode, TypeSize &Scale,
...
-    MaxOffset = 1008;
+    MinOffset = -64;
+    MaxOffset = 63;
     break;
   case AArch64::STPXpre:
   case AArch64::LDPXpost:
   case AArch64::STPDpre:
   case AArch64::LDPDpost:
     Scale = TypeSize::getFixed(8);
     Width = TypeSize::getFixed(8);
```
[the remark this answers is not in the document]
Comment (davemgreen) [reply llvm/lib/Target/AArch64/AArch64InstrInfo.cpp]: Yeah that sounds good, I forgot to change that one. Anything else you think should change?

[the remark this answers is not in the document]
Comment (romainthomas) [reply llvm/lib/Target/AArch64/AArch64InstrInfo.cpp]: Nop looks good :+1:

**Review by efriedma-quic — APPROVED** [review]
LGTM

It's a little surprising how broken this was without anyone noticing...

File: llvm/lib/Target/AArch64/AArch64InstrInfo.cpp [hunk 2]
```diff
@@ -3722,26 +3945,26 @@ bool AArch64InstrInfo::getMemOpInfo(unsigned Opcode, TypeSize &Scale,
...
     MaxOffset = 63;
     break;
   // pre/post inc
   case AArch64::STPQpre:
   case AArch64::LDPQpost:
     Scale = TypeSize::getFixed(16);
-    Width = TypeSize::getFixed(16);
-    MinOffset = -1024;
-    MaxOffset = 1008;
+    Width = TypeSize::getFixed(16 * 2);
```
Comment (sdesmalen-arm) [inline llvm/lib/Target/AArch64/AArch64InstrInfo.cpp:3956]: nit: these case statement can be combined with the ones for `LDPQi`, etc?

File: llvm/lib/Target/AArch64/AArch64InstrInfo.cpp [hunk 3]
```diff
@@ -3722,26 +3945,26 @@ bool AArch64InstrInfo::getMemOpInfo(unsigned Opcode, TypeSize &Scale,
...
     break;
   case AArch64::STPXpre:
   case AArch64::LDPXpost:
   case AArch64::STPDpre:
   case AArch64::LDPDpost:
     Scale = TypeSize::getFixed(8);
-    Width = TypeSize::getFixed(8);
-    MinOffset = -512;
-    MaxOffset = 504;
+    Width = TypeSize::getFixed(8 * 2);
```
Comment (sdesmalen-arm) [inline llvm/lib/Target/AArch64/AArch64InstrInfo.cpp:3965]: nit: Same for these, but then wrt `LDPXi`

**Review by sdesmalen-arm — APPROVED** [review]
Two little nits, but otherwise LGTM

**Pull request merged**